Sources/NIOHTTPResponsiveness/SimpleResponsivenessRequestMux.swift

```diff
@@ -99,26 +135,24 @@ public final class SimpleResponsivenessRequestMux: ChannelInboundHandler {
                     self.addHandlerOrInternalError(context: context, handler: HTTPDrippingDownloadHandler())
                 }
             default:
-                self.writeSimpleResponse(
-                    context: context,
-                    status: .notFound,
-                    body: SimpleResponsivenessRequestMux.notFoundBody
-                )
+                if !self.forwardOtherRequests {
+                    self.writeSimpleResponse(
+                        context: context,
+                        status: .notFound,
+                        body: SimpleResponsivenessRequestMux.notFoundBody
+                    )
+                    return
+                }
             }
         }
 
-        // Only pass through data through if we've actually added a handler. If we didn't add a handler, it's because we
-        // directly responded. In this case, we don't care about the rest of the request.
-        if self.handlerAdded {
-            context.fireChannelRead(data)
-        }
+        context.fireChannelRead(data)
```

Comment:
This can fire if adding a handler fails. That's likely to be logically wrong, as this handler will have already written a 500 error, so we mustn't forward it on. The easiest way to tackle that is probably to have `addHandlerOrInternalError` become throwing, and then handle the internal error at the `channelRead` scope.